addons/rose/addon/modifiers/on-click-inside.js

```diff
@@ -5,16 +5,19 @@ import { modifier } from 'ember-modifier';
  * This is useful especially for dropdowns to toggle content visibility
  * when the user clicks an item.
  */
-export default modifier((element, [callback]) => {
-  function handleClick(event) {
-    if (element.contains(event.target)) {
-      callback(element, event);
+export default modifier(
+  (element, [callback]) => {
+    function handleClick(event) {
+      if (element.contains(event.target)) {
```

Comment:
I know this was present previously and unrelated to your change, but why is this modifier needed?

How is this different from the default `{{on "click"}}` modifier? Why do we need to check if the element contains the target, are there scenarios where we could click an element that isn't under the DOM hierarchy of where we attached this modifier?

Reply:
Unfortunately I can't answer your question @ZedLi .

As you said the modifiers were there, the goal and the scope for this PR was purely to just update the ember addon that allow those modifiers to work.

That said, I agree that these questions should be answered. So [I created this tech debt ticket for it](https://hashicorp.atlassian.net/browse/ICU-6387) (feel free to take it).

The files that have those modifiers have comments mentioning is useful for dropdowns and such, maybe we can start checking there.

Reply:
Cool! And I was just asking for the `on-click-inside` modifier, the `on-click-outside` modifier makes perfect sense to me and is what I would expect

Reply:
@ZedLi Awesome!! Updated the ticket making reference to check just `on-click-inside` 😄 Thanks for the catch!